### PR TITLE
LPS-134898

### PR DESCRIPTION
**Pull request by marianoalvarosaiz**

### CODE REVIEW

File: modules/apps/expando/expando-test/src/testIntegration/java/com/liferay/expando/service/test/ExpandoValueLocalServiceTest.java [hunk 1]
```diff
@@ -187,6 +194,58 @@ public void testAddStringArrayValue() throws Exception {
...
+		ExpandoTestUtil.addValue(
+			_expandoTable, column1, classPK, "column1-one");
+		ExpandoTestUtil.addValue(
+			_expandoTable, column2, classPK, "column2-one");
+
+		// Remove cache to force  ExpandoValuePeristence find methods be
+		// executed against the database to force flush in between.
+
+		_entityCache.clearCache(ExpandoValueImpl.class);
+		_finderCache.clearCache(ExpandoValueImpl.class);
```
Comment (dantewang) [inline modules/apps/expando/expando-test/src/testIntegration/java/com/liferay/expando/service/test/ExpandoValueLocalServiceTest.java:218]: @marianoalvarosaiz I see here you need to clean up the cache to make sure the later method invocation does the re-fetch which leads to a query that flushes the session. But is this happening in one of the reportings? Like when importing data?

**Comment by marianoalvarosaiz** [comment]
It does happen with Andre Fabbro use case. His reproduction steps are:

1. Create some User custom fields.
2. Create a new Authenticator that updates the custom fields when logging in.
3. If a try to log in I get no error, but if I restart the server or clean the cache I get an error when I log.

File: modules/apps/expando/expando-test/src/testIntegration/java/com/liferay/expando/service/test/ExpandoValueLocalServiceTest.java (hunk 1 above)
Comment (marianoalvarosaiz) [reply modules/apps/expando/expando-test/src/testIntegration/java/com/liferay/expando/service/test/ExpandoValueLocalServiceTest.java:218]: @dantewang 
It does happen with Andre Fabbro use case. His reproduction steps are:

    Create some User custom fields.
    Create a new Authenticator that updates the custom fields when logging in.
    If a try to log in I get no error, but if I restart the server or clean the cache I get an error when I log.

Comment (marianoalvarosaiz) [reply modules/apps/expando/expando-test/src/testIntegration/java/com/liferay/expando/service/test/ExpandoValueLocalServiceTest.java:218]: But is the same with other LPP, if you wait enough time after creating the first expando values and you have two different expando columns whose value is going to be modified sequentially you get the same exception.

**Comment by catarineaguiar** [comment]
ci:stop:relevant

**Comment by catarineaguiar** [comment]
ci:test:relevant

**Comment by catarineaguiar** [comment]
ci:test:relevant

**Comment by zhangyan0701** [comment]
ci:test:relevant

**Comment by catarineaguiar** [comment]
@marianoalvarosaiz no regressions found.

**Comment by marianoalvarosaiz** [comment]
See https://github.com/liferay-core-infra/liferay-portal/pull/2383

**Pull request closed without merging**